.eslintrc

```diff
@@ -28,6 +28,7 @@
     "no-confusing-arrow": "off",
     "no-plusplus": "off",
     "no-unused-expressions": "off",
+    "no-use-before-define": "off",
```

Comment:
where did this become an issue?

Reply:
> where did this become an issue?

It was specifically with this reference to `getValidationErrors`:

https://github.com/18F/identity-idp/blob/e61e74fb972e11cf1f90cc542da27867d5eb2288/app/javascript/packages/document-capture/components/form-steps.jsx#L119

Technically, `getValidationErrors` could be moved up to occur before this `useEffect` where it's referenced, but...

1. This isn't always possible, in cases where an early return would force a specific arrangement of hooks and variable / function definitions, or in interdependencies between functions
2. For readability and to abide by [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html), conventionally I'd expect hooks to be arranged together toward the top of a function component.

Since some components (including `FormHooks`) will have early return values, it can be be beneficial to only assign values in the paths where it's guaranteed to be referenced (as late as possible, [see related](https://github.com/WordPress/gutenberg/pull/12828)). In practice, this usually means hooks and their dependencies before the early return, and everything else after. `getValidationErrors` straddles the line, since it's used both by a hook and the submit handler. That being said, as I write this it occurs to me that there's some fragility here in that the function could potentially not be defined if the early return would be triggered (this wouldn't actually happen). It should probably be defined at least before the `if (!step) {` condition, if not before the hook.

Reply:
neat, thanks for the Rules of Hooks reference 👍

Reply:
I opted to just define the function before `useEffect` and revert the ESLint change in 3f035d5fc.

We might run into this again at some point in the future, but we can deal with it if and when that time comes.